### PR TITLE
Add human readable argument converter names

### PR DESCRIPTION
**Pull request by OoLunar**
# Summary
Paves way to a decent help command.

# Notes
Tested.

### CODE REVIEW

**Review by Plerx2493 — APPROVED** [review]

File: DSharpPlus.Commands/Converters/TimeSpanConverter.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public partial class TimeSpanConverter : ISlashArgumentConverter<TimeSpan>, ITex
     private static partial Regex _getTimeSpanRegex();
 
     public DiscordApplicationCommandOptionType ParameterType { get; init; } = DiscordApplicationCommandOptionType.String;
+    public string ReadableName { get; init; } = "Time Duration";
```
Comment (akiraveliara) [inline DSharpPlus.Commands/Converters/TimeSpanConverter.cs:18]: Duration should be enough

File: DSharpPlus.Commands/Converters/DateTimeOffsetConverter.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@ namespace DSharpPlus.Commands.Converters;
 public class DateTimeOffsetConverter : ISlashArgumentConverter<DateTimeOffset>, ITextArgumentConverter<DateTimeOffset>
 {
     public DiscordApplicationCommandOptionType ParameterType { get; init; } = DiscordApplicationCommandOptionType.String;
+    public string ReadableName { get; init; } = "Date and Time";
```
Comment (akiraveliara) [inline DSharpPlus.Commands/Converters/DateTimeOffsetConverter.cs:14]: Timestamp?

**Review by akiraveliara — CHANGES_REQUESTED** [review]
why aren't the other properties static abstract and/or get-only, on that note?

also, are we sure we want to Discord-prefix everything - i feel like `File` or `Message Link` is clear enough; and are we sure we want to expose the names of C# keywords to users? `Integer (minvalue - maxvalue)` is probably enough, imo

File: DSharpPlus.Commands/Converters/IArgumentConverter.cs [hunk 3]
```diff
@@ -4,7 +4,10 @@ namespace DSharpPlus.Commands.Converters;
 using DSharpPlus.AsyncEvents;
 using DSharpPlus.Entities;
 
-public interface IArgumentConverter { }
+public interface IArgumentConverter
+{
+    public string ReadableName { get; init; }
```
[the remark this answers is not in the document]
Comment (OoLunar) [reply DSharpPlus.Commands/Converters/IArgumentConverter.cs:9]: Static abstract doesn't support init.

[the remark this answers is not in the document]
Comment (akiraveliara) [reply DSharpPlus.Commands/Converters/IArgumentConverter.cs:9]: why do they need `init`

File: DSharpPlus.Commands/Converters/DateTimeOffsetConverter.cs (hunk 2 above)
Comment (OoLunar) [reply DSharpPlus.Commands/Converters/DateTimeOffsetConverter.cs:14]: Too generalized for the common discord user

**Comment by OoLunar** [comment]
> why aren't the other properties static abstract and/or get-only, on that note?

Get only means the value can change at any point in time. Besides that, I'm not particularly against static abstract

> also, are we sure we want to Discord-prefix everything - i feel like `File` or `Message Link` is clear enough

Yes we do want to prefix Discord argument converters with Discord to disambiguate other custom argument converters.

> and are we sure we want to expose the names of C# keywords to users? `Integer (minvalue - maxvalue)` is probably enough, imo

We should try to provide the most amount of information available on a best effort basis. The user can always override the readable name themselves in their own implementation through pattern matching

File: DSharpPlus.Commands/Converters/IArgumentConverter.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (OoLunar) [reply DSharpPlus.Commands/Converters/IArgumentConverter.cs:9]:
```
DSharpPlus.Commands/Processors/TextCommands/TextCommandProcessor.cs(14,21): error CS8920: The interface 'ITextArgumentConverter' cannot be used as type argument. Static member 'IArgumentConverter.ReadableName' does not have a most specific implementation in the interface.
```

[the remark this answers is not in the document]
Comment (akiraveliara) [reply DSharpPlus.Commands/Converters/IArgumentConverter.cs:9]: that seems rather silly, but, fine - ill look into that myself at some point

**Pull request merged**